src/app/features/ledger/utils/generic-ledger-utils.ts

```diff
@@ -48,3 +61,30 @@ export function prepareLedgerDeviceForAppFn<T extends () => Promise<unknown>>(co
     return app;
   };
 }
+
+export async function promptOpenAppOnDevice(appName: string) {
+  const tmpTransport = await TransportWebUSB.create();
+  const tmpBitcoinApp = new BitcoinApp(tmpTransport);
+  const resp = await tmpBitcoinApp.getAppAndVersion();
+  if (resp.name !== appName && resp.name !== LEDGER_APPS_MAP.MAIN_MENU) {
+    throw new Error(LedgerConnectionErrors.IncorrectAppOpened);
+  }
+  if (resp.name !== appName) {
+    await tmpTransport.send(0xe0, 0xd8, 0x00, 0x00, Buffer.from(appName, 'ascii'));
+  }
+  // for some reason sending open app buffer to ledger will close the connection afterwards.
+  // we need to add a delay for this transport to properly finish for another one to open.
+  await delay(500);
```

Comment:
How could we do this in a way where we don't depend on a timeout?

Reply:
`transport.close` also throws an error for some reason, and it seems like transport was closed by itself after a `send` operation. Opening another transport without a delay produces an error too. I've not completely figured out this part, but adding 500ms delay after a `send` and opening another transport afterwards results in a working transport :D